agreement-service.js

```diff
@@ -23,7 +23,13 @@ function factory($http, config) {
     });
     // TODO: make url configurable
     var url = config.data.baseUri + '/agreements/accepted';
-    return Promise.resolve($http.post(url, {agreement: agreementUrls}));
+    return Promise.resolve($http.post(url, {agreement: agreementUrls}))
+      .then(function(response) {
+        if(response.status !== 201) {
+          throw new Error('Agreement acceptance was not properly recorded.');
```

Comment:
This is super user-facing, but the text we want is probably something like:

`There was a problem recording the acceptance of the agreement. Please try again later or contact support.`